identifier/identifier.go

```diff
@@ -75,3 +80,70 @@ func NewIP(ip netip.Addr) ACMEIdentifier {
 		Value: ip.String(),
 	}
 }
+
+// FromCert extracts the Subject Alternative Names from a certificate, and
+// returns a slice of ACMEIdentifiers and an error. It does not extract the
+// Subject Common Name.
+//
+// FromCSR is similar but handles CSRs, and is kept separate so that it's always
+// clear we are handling an untrusted CSR.
+func FromCert(cert *x509.Certificate) []ACMEIdentifier {
+	var sans []ACMEIdentifier
+	for _, name := range cert.DNSNames {
+		sans = append(sans, NewDNS(name))
+	}
+
+	for _, ip := range cert.IPAddresses {
+		sans = append(sans, ACMEIdentifier{
+			Type:  TypeIP,
+			Value: ip.String(),
+		})
+	}
+
+	return Normalize(sans)
+}
+
+// FromCSR extracts the Subject Common Name and Subject Alternative Names from a
+// CSR, and returns a slice of ACMEIdentifiers and an error.
+//
+// FromCert is similar but handles certs, and is kept separate so that it's
+// always clear we are handling an untrusted CSR.
```

Comment:
nit: The first half of this sentence is appropriately flip-flopped from the corresponding sentence in FromCert's comment. But then the rest of it is left the same, which feels weird. "FromCert is kept separate so that it's always clear we are handling an untrusted CSR" -- but FromCert isn't handling an untrusted CSR! Maybe this is nothing but it threw my brain for a loop.